packages/sdk-core/src/modules/database/BacktraceDatabase.ts

```diff
@@ -115,23 +125,63 @@ export class BacktraceDatabase implements BacktraceModule {
     public add(
         backtraceData: BacktraceData,
         attachments: BacktraceAttachment<unknown>[],
-    ): BacktraceDatabaseRecord | undefined {
+    ): ReportBacktraceDatabaseRecord | undefined {
         if (!this._enabled) {
             return undefined;
         }
 
-        this.prepareDatabase();
+        const sessionId = backtraceData.attributes?.['application.session'];
 
-        const record: BacktraceDatabaseRecord = {
-            count: 1,
+        const record: ReportBacktraceDatabaseRecord = {
+            type: 'report',
             data: backtraceData,
             timestamp: TimeHelper.now(),
             hash: '',
             id: IdGenerator.uuid(),
             locked: false,
             attachments: attachments,
+            sessionId: typeof sessionId === 'string' ? sessionId : undefined,
         };
 
+        this.prepareDatabase([record]);
+        const saveResult = this._storageProvider.add(record);
+        if (!saveResult) {
+            return undefined;
+        }
+
+        this._databaseRecordContext.add(record);
+        this.lockSessionWithRecord(record);
+
+        return record;
+    }
+
+    /**
+     * Adds Bactrace attachment to the database
+     * @param backtraceData diagnostic data object
+     * @param attachments attachments
+     * @returns record if database is enabled. Otherwise undefined.
+     */
+    public addAttachment(
+        rxid: string,
+        attachment: BacktraceAttachment,
+        sessionId?: string,
+    ): AttachmentBacktraceDatabaseRecord | undefined {
+        if (!this._enabled) {
+            return undefined;
+        }
+
+        const record: AttachmentBacktraceDatabaseRecord = {
+            type: 'attachment',
+            timestamp: TimeHelper.now(),
+            hash: '',
+            id: IdGenerator.uuid(),
+            rxid,
+            locked: false,
+            attachment,
+            sessionId: typeof sessionId === 'string' ? sessionId : undefined,
```

Comment:
can we add attachment without sessionid?

Reply:
I guess not. I made it mandatory.